### PR TITLE
Add service dependency check

### PR DESCRIPTION
**Pull request by rauhut**

### CODE REVIEW

**Review by rocktavious — CHANGES_REQUESTED** [review]
Looks like 2 copy/paste things that need to be cleaned up.

And the tests are failing because the opslevel-go dependency doesn't have your new stuff yet.  You can fix this in this branch by modifying the `go.mod` file at the bottom there is a `replace` directive you need to uncomment out and commit that.  You will also need to make sure the submodule is pointed to your WIP commit and commit the update that happens to gitmodules to this branch.

Let me know if you have trouble - otherwise we'll have to wait until the upstream PR is merged.

File: opslevel/resource_opslevel_check_service_dependency.go
Comment (rauhut) [reply opslevel/resource_opslevel_check_service_dependency.go]: 👍  I noticed there were actually two other places where we actually used that same string incorrectly (for non tool usage checks), so I updated those descriptions as well in my latest commit

Comment (rocktavious) [reply opslevel/resource_opslevel_check_service_dependency.go]: ❤️

**Review by rocktavious — APPROVED** [review]
LGTM

**Pull request merged**
